### ISSUE TITLE
Calibrate inference_sim_floor to live Δ for retrieval-side left-censoring fix

### ISSUE THREAD

**Issue by samscarrow** [git-scarrow/fast-associative-memory#74]
## Context

Issue #73 implemented `RelativeVigilance`, which adapts the write gate to the live off-class similarity distribution. The four-run evaluation (synthetic × {margin, relative} and real × {margin, relative}) confirmed that `RelativeVigilance` improves write-gate calibration — it suppresses `rho_probe` by ~10–15% on real embeddings — but **does not move blend onset off epoch 0** on the real manifold.

The left-censoring problem is a retrieval phenomenon, not a write-gate phenomenon.

## Root cause

On the real 20NG / all-MiniLM-L6-v2 manifold, Δ≈0.422 and off-class ρ≈0.233 at rest. With `inference_temp=0.05` and ~96 off-class slots in memory at startup, the retrieval softmax already puts majority mass on off-class prototypes before any drift is applied:

```
exp(0.422 / 0.05) ≈  4,600   per within-class slot
exp(0.233 / 0.05) ≈    106   per off-class slot × 96 slots ≈ 10,200 total
```

The aggregate off-class vote mass (~69%) exceeds `blend_eps=0.10` from epoch 0. No write-gate change can fix this because the write gate only controls which slot is updated, not the retrieval softmax distribution.

## Proposed fix

Add a **live Δ-relative `inference_sim_floor`** that masks out retrieval candidates below a threshold anchored to the observed within-class similarity:

```python
sim_floor = delta_live - k * inference_temp
```

where `delta_live` is the running EMA of within-class self-similarity (already available via `probe_cross_class_similarity()` / `get_stats()`) and `k` is a tunable number of logit steps. Only prototypes within `k` softmax temperature steps of the best within-class match contribute to the vote.

This is the retrieval-side analogue of `RelativeVigilance`: where `RelativeVigilance` sets the write gate relative to the off-class distribution, a live `sim_floor` sets the retrieval gate relative to the within-class distribution.

`ContinuousCAM` already has an `inference_sim_floor` field (currently a static float, default 0.0). The change is to make it dynamically updated from a live Δ estimate.

## Proposed approach

1. Add a `RetrievalFloorPolicy` (or equivalent) that owns:
   - `ema_beta: float` — smoothing coefficient for the running Δ estimate
   - `k_logit_steps: float` — how many temp steps below Δ to place the floor
   - `floor_min: float` — safety lower bound
   - `update(delta_obs: float)` — EMA update from observed within-class sim
   - `floor() -> float` — current floor value = `clamp(delta_ema - k * temp, floor_min, 1.0)`

2. Wire it into `ContinuousCAM.forward()` and `probe_cross_class_similarity()` as an optional policy (analogous to `dynamic_vigilance`), replacing the static `inference_sim_floor`.

3. Update `benchmarks/probe_contraction.py` with `--retrieval-floor-policy {static,live-delta}` and measure whether blend onset moves off epoch 0 on the real manifold.

## Evaluation

Reuse `benchmarks/probe_contraction.py` with real 20NG + all-MiniLM-L6-v2. Primary success criterion:

- **blend onset epoch > 0 on the real manifold** (left-censoring resolved)

Secondary metrics (compared against `--retrieval-floor-policy static` baseline):
- observed blend ρ at onset
- frac_blended trajectory across epochs
- off-class vote mass trajectory
- memory occupancy / allocation rate
- accuracy / label purity

## Scope

Write-gate policy (`RelativeVigilance` from #73) is orthogonal. The two can compose: `RelativeVigilance` for the write side, live Δ floor for the retrieval side. Do not entangle them.
